lib/rules/line-comment-position.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @fileoverview Rule to enforce the position of line comments
+ * @author Alberto Rodríguez
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "enforce position of line comments",
+            category: "Stylistic Issues",
+            recommended: false
+        },
+
+        schema: [
+            {
+                oneOf: [
+                    {
+                        enum: ["above", "beside"]
+                    },
+                    {
+                        type: "object",
+                        properties: {
+                            position: {
+                                enum: ["above", "beside"]
+                            },
+                            ignorePattern: {
+                                type: "string"
+                            },
+                            applyDefaultPatterns: {
+                                type: "boolean"
+                            }
+                        },
+                        additionalProperties: false
+                    }
+                ]
+            }
+        ]
+    },
+
+    create(context) {
+        const DEFAULT_IGNORE_PATTERN = "^\\s*(?:eslint|jshint\\s+|jslint\\s+|istanbul\\s+|globals?\\s+|exported\\s+|jscs|falls?\\s?through)";
```

Comment:
Sure, but it's internal. Using a string here results in a performance hit,
and I'm not convinced that it's more readable either.

On Aug 25, 2016 5:31 PM, "alberto" <notifications@github.com> wrote:

> In lib/rules/line-comment-position.js
> <https://github.com/eslint/eslint/pull/6953#discussion_r76337140>:
>
> > +                            ignorePattern: {
> > +                                type: "string"
> > +                            },
> > +                            applyDefaultPatterns: {
> > +                                type: "boolean"
> > +                            }
> > +                        },
> > +                        additionalProperties: false
> > +                    }
> > +                ]
> > +            }
> > +        ]
> > +    },
> > +
> > +    create(context) {
> > +        const DEFAULT_IGNORE_PATTERN = "^\\s*(?:eslint|jshint\\s+|jslint\\s+|istanbul\\s+|globals?\\s+|exported\\s+|jscs|falls?\\s?through)";
>
> I used a string for symmetry with ignorePattern.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/eslint/eslint/pull/6953/files/55e1849ca32affa2fb639624d14c153cdbd470d0#r76337140>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARWejQEM9RWQjG8h4TAyuPn6uZB_Dh7ks5qjhe-gaJpZM4JpTWN>
> .
>